### Importar bibliotecas

In [34]:
import pandas as pd
import matplotlib as plt
import numpy as np

### Carga de datos formato JSON

In [3]:
laptop = pd.read_parquet('C:/Users/sr_tu/Desktop/bootcamp/DS102024/3-MachineLearning/2-Supervisado/1-Clasificacion/Ejercicio/train.parquet')

### Primeros pasos

In [4]:
laptop.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 912 entries, 0 to 911
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   laptop_ID         912 non-null    int64  
 1   Company           912 non-null    object 
 2   Product           912 non-null    object 
 3   TypeName          912 non-null    object 
 4   Inches            912 non-null    object 
 5   ScreenResolution  912 non-null    object 
 6   Cpu               912 non-null    object 
 7   Ram               912 non-null    object 
 8   Memory            912 non-null    object 
 9   Gpu               912 non-null    object 
 10  OpSys             912 non-null    object 
 11  Weight            912 non-null    object 
 12  Price_euros       912 non-null    float64
dtypes: float64(1), int64(1), object(11)
memory usage: 92.8+ KB


### El fichero se compone de 912 lineas y 13 columnas y ningun valor nulo

In [5]:
laptop.tail()

laptop_ID Company                               Product  \
907        837    Acer                         Chromebook 11   
908        626    Acer  CB5-132T-C9KK (N3160/4GB/32GB/Chrome   
909        225    Dell                         Inspiron 3576   
910         17    Dell                         Inspiron 3567   
911         87    Dell                         Inspiron 7577   

               TypeName Inches                ScreenResolution  \
907             Netbook   11.6              IPS Panel 1366x768   
908  2 in 1 Convertible   11.6  IPS Panel Touchscreen 1366x768   
909            Notebook   15.6               Full HD 1920x1080   
910            Notebook   15.6               Full HD 1920x1080   
911              Gaming   15.6     IPS Panel Full HD 1920x1080   

                                      Cpu   Ram                Memory  \
907  Intel Celeron Dual Core N3060 1.6GHz   4GB    32GB Flash Storage   
908  Intel Celeron Quad Core N3160 1.6GHz   4GB    32GB Flash Storage   
909            Intel Core i5 8250U 1.6GHz   8GB             256GB SSD   
910            Intel Core i7 7500U 2.7GHz   8GB             256GB SSD   
911           Intel Core i7 7700HQ 2.8GHz  16GB  128GB SSD +  1TB HDD   

                            Gpu       OpSys  Weight  Price_euros  
907       Intel HD Graphics 400   Chrome OS  1.35kg       355.00  
908       Intel HD Graphics 400   Chrome OS  1.25kg       379.00  
909              AMD Radeon 520       Linux   2.2kg       677.35  
910          AMD Radeon R5 M430  Windows 10   2.2kg       745.00  
911  Nvidia GeForce GTX 1050 Ti  Windows 10  2.65kg      1349.00

In [6]:
laptop.describe()

laptop_ID  Price_euros
count   912.000000   912.000000
mean    663.118421  1111.262533
std     380.713558   682.109694
min       1.000000   174.000000
25%     339.750000   598.742500
50%     666.500000   977.000000
75%     985.250000  1472.650000
max    1320.000000  5499.000000

In [7]:
laptop

laptop_ID  Company                                  Product  \
0          599     Dell                            Inspiron 3179   
1           92     Asus  FX550IK-DM018T (FX-9830P/8GB/1TB/Radeon   
2          744       HP                                 ZBook 15   
3          729       HP                            Pavilion X360   
4          957  Toshiba                          Tecra A50-C-1ZV   
..         ...      ...                                      ...   
907        837     Acer                            Chromebook 11   
908        626     Acer     CB5-132T-C9KK (N3160/4GB/32GB/Chrome   
909        225     Dell                            Inspiron 3576   
910         17     Dell                            Inspiron 3567   
911         87     Dell                            Inspiron 7577   

               TypeName Inches                ScreenResolution  \
0    2 in 1 Convertible   11.6            Touchscreen 1366x768   
1                Gaming   15.6               Full HD 1920x1080   
2           Workstation   15.6               Full HD 1920x1080   
3    2 in 1 Convertible   13.3  IPS Panel Touchscreen 1366x768   
4              Notebook   15.6                        1366x768   
..                  ...    ...                             ...   
907             Netbook   11.6              IPS Panel 1366x768   
908  2 in 1 Convertible   11.6  IPS Panel Touchscreen 1366x768   
909            Notebook   15.6               Full HD 1920x1080   
910            Notebook   15.6               Full HD 1920x1080   
911              Gaming   15.6     IPS Panel Full HD 1920x1080   

                                      Cpu   Ram                Memory  \
0                Intel Core M 7Y30 1.0GHz   4GB             128GB SSD   
1                       AMD FX 9830P 3GHz   8GB               1TB HDD   
2             Intel Core i7 7700HQ 2.8GHz  16GB             256GB SSD   
3              Intel Core i5 7200U 2.5GHz   8GB             256GB SSD   
4              Intel Core i5 6200U 2.3GHz   8GB             256GB SSD   
..                                    ...   ...                   ...   
907  Intel Celeron Dual Core N3060 1.6GHz   4GB    32GB Flash Storage   
908  Intel Celeron Quad Core N3160 1.6GHz   4GB    32GB Flash Storage   
909            Intel Core i5 8250U 1.6GHz   8GB             256GB SSD   
910            Intel Core i7 7500U 2.7GHz   8GB             256GB SSD   
911           Intel Core i7 7700HQ 2.8GHz  16GB  128GB SSD +  1TB HDD   

                            Gpu       OpSys  Weight  Price_euros  
0         Intel HD Graphics 615  Windows 10  1.39kg       603.00  
1             AMD Radeon RX 560  Windows 10  2.45kg       699.00  
2           Nvidia Quadro M2200  Windows 10   2.6kg      2419.00  
3         Intel HD Graphics 620  Windows 10  1.58kg       836.60  
4         Intel HD Graphics 520  Windows 10   2.2kg      1119.00  
..                          ...         ...     ...          ...  
907       Intel HD Graphics 400   Chrome OS  1.35kg       355.00  
908       Intel HD Graphics 400   Chrome OS  1.25kg       379.00  
909              AMD Radeon 520       Linux   2.2kg       677.35  
910          AMD Radeon R5 M430  Windows 10   2.2kg       745.00  
911  Nvidia GeForce GTX 1050 Ti  Windows 10  2.65kg      1349.00  

[912 rows x 13 columns]

### Filtrado por tipo de PC´S 

In [8]:
conteo_tipos = laptop['TypeName'].value_counts()
print (conteo_tipos)

TypeName
Notebook              498
Gaming                149
Ultrabook             132
2 in 1 Convertible     87
Workstation            24
Netbook                22
Name: count, dtype: int64


In [11]:
TypeName = laptop['TypeName'].unique()
Marca = laptop['Company'].unique()

### Filtrado por Marcas

In [12]:
Marca = laptop['Company'].unique()

In [13]:
conteo_marcas = laptop['Company'].value_counts()
print (conteo_marcas)

Company
Dell         214
Lenovo       206
HP           184
Asus         105
Acer          72
MSI           44
Toshiba       35
Apple         13
Samsung        9
Razer          5
Mediacom       5
Google         3
LG             3
Microsoft      3
Fujitsu        3
Xiaomi         2
Huawei         2
Chuwi          2
Vero           2
Name: count, dtype: int64


### Filtrado por procesadores

In [14]:
laptop['Cpu'].str.contains('Intel').value_counts()
#laptop['Cpu'].str.contains('AMD').value_counts()

Cpu
True     866
False     46
Name: count, dtype: int64

### Filtrado por memoria Ram

In [15]:
MemoriRam = laptop['Ram'].unique()
Ram = laptop['Ram'].unique()

In [16]:
Ram

array(['4GB', '8GB', '16GB', '6GB', '12GB', '2GB', '32GB', '24GB', '64GB'],
      dtype=object)

### Filtrado por sistema operativo 

In [17]:
sistemas_op = laptop['OpSys'].value_counts()
print (sistemas_op)

OpSys
Windows 10      760
Linux            43
No OS            39
Windows 7        31
Chrome OS        20
Mac OS X          7
macOS             6
Windows 10 S      5
Android           1
Name: count, dtype: int64


In [40]:
#pulgadas = laptop['Inches'].unique()
pulgadas = laptop['Inches'].value_counts()

In [41]:
print(pulgadas)

Inches
15.6    461
14.0    131
17.3    118
13.3    117
12.5     28
11.6     25
12.3      5
13.9      5
12.0      5
15.0      4
10.1      3
13.5      3
13.0      2
15.4      1
18.4      1
11.3      1
14.1      1
17.0      1
Name: count, dtype: int64
